src/main/dynamo/operations/batchGetItems.ts

```diff
@@ -48,3 +45,19 @@ export async function batchGetItems<T extends PartitionAndSortKey<TaggedModel>>(
     return { items: [], unprocessedKeys: [] }
   }
 }
+
+function screenKeysForDuplicates<T extends PartitionAndSortKey<TaggedModel>>(keys: T[], table: Table<string, string>) {
```

Comment:
suggestion for better name:  `mapToKeysAndRemoveDuplicate`

Reply:
Could you also add an explicit return type?

Reply:
Why not return an error if duplicate keys are provided?   This could easily turn into undiscovered bugs if we silently discard key-value associations.... especially since the current behavior is not "last entry wins".    I think it might be best to raise an error.

Reply:
> Why not return an error if duplicate keys are provided? This could easily turn into undiscovered bugs if we silently discard key-value associations.... especially since the current behavior is not "last entry wins".

For writes, that makes sense to me. For the case here, we're deduplicating keys that are used for reads, so I think the deduplication without raising an error is appropriate.

Could you share more about the types of bugs you're thinking we might hit if deduplicating keys for reads?

Reply:
Ah sorry, my mind was assuming writes.  I don't see a problem for reads, though I'd consider it a smell... why would we have duplicates?   

If we don't want to fail, it might be a good idea to log a warning of some sort.   I'm in the camp that it's okay to be helpful and convenient but it's typically better to bubble up any anomalous/suspicious use of an API.   

(This isn't a recommendation, it's a suggestion.  Take it as "one engineer's opinion", not "CTO made me do it!" ;)

Reply:
> If we don't want to fail, it might be a good idea to log a warning of some sort. I'm in the camp that it's okay to be helpful and convenient but it's typically better to bubble up any anomalous/suspicious use of an API.

Cool, that makes sense. The background on this is that @teeeteee found that ClinHub was getting errors from Vault about duplicate keys when making certain `getVaultItemsByTags` queries. I think he was looking into the reason the client (ClinHub) is sending duplicate keys in that query but took this approach as a first step since it was easier to address and handles the issue across clients.

Logging a warning does sound like a good idea since sending dups could be indicative of some other issue in that client. 👍

Reply:
Popping my head up from my ostrich hole for a second... :)

This is fine (for now). But,  I think the (long-term) approach we want here is take the simple route and just let this library throw an error if/when duplicate keys are encountered. 

In the same vein as Alex's comment, if a caller does something it shouldn't, I find it preferable for libraries to just throw errors vs attempt to save the caller from themselves. This results in cleaner libraries, with less bespoke code. 

Duplicate keys here is indicative of a bug in the caller's code. So (ideally) we'd just fix the issue in the caller's code. Since this is already done/merged -- I think it's ok to wait until I get back to address that.